src/main/java/mate/academy/dao/jdbc/CarDaoJdbc.java

```diff
@@ -0,0 +1,185 @@
+            statementForCar.setLong(2, car.getManufacturer().getId());
+            statementForCar.executeUpdate();
+            ResultSet resultSet = statementForCar.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create car: " + car, e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getById = "SELECT * FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.manufacturer_id "
+                + "WHERE c.car_id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statementCar = connection.prepareStatement(getById)) {
+            statementCar.setLong(1, id);
+            ResultSet resultSet = statementCar.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car by id: " + id, e);
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllCar = "SELECT * FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.manufacturer_id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getAllCar)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get list of car", e);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCars = "UPDATE cars SET car_model = ?, manufacturer_id = ? "
+                + "WHERE car_id = ? and is_deleted = FALSE;";
+        String deleteCarsDrivers = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(updateCars);
+                PreparedStatement statementDelete =
+                        connection.prepareStatement(deleteCarsDrivers)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+            statementDelete.setLong(1, car.getId());
+            statementDelete.executeUpdate();
+            createCarsDrivers(car);
```

Comment:
move this line out of the `try` block - at the moment you opened one connection inside of another